### DISCUSSION TITLE
SplineCNN parameters in main paper

### DISCUSSION THREAD

**Discussion by mfarazi1991** [pyg-team/pytorch_geometric#4624] [Q&A] [answered]
Hi Matthias,

I am doing a benchmark for shape correspondence and I am not sure how to convert the parameters in SplineCNN paper to SplineConv parameters. **SplineConv(in_channels, out_channels, dim, kernel_size)** is the PyG conv layer, but I am not sure how to define dim and kernel size (the rest is straightforward). Do you have the code for any experiments like graph node classification or others available? I want to see the network setup if possible. Thanks!

**Comment by rusty1s** [comment, marked as answer]
You can find an example of SplineCNN here: https://github.com/pyg-team/pytorch_geometric/blob/master/examples/faust.py

* `dim` defines the feature dimensionality of edge features. For example, in a point cloud setting with edge features denoting spatial relations, this is set to `3`.
* `kernel:size` defines the number of parameters we want to learn in each spatial dimension. This is similar to the kernel size of traditional CNNs. If set to `5` with `dim=3`, one will apply a kernel window of `5x5x5` parameters.

**Reply by mfarazi1991** [reply]
Like always, thank you for your prompt and helpful response.
